pkg/detectors/detectors.go

```diff
@@ -26,6 +26,22 @@ type Detector interface {
 	Type() detectorspb.DetectorType
 }
 
+// CustomResultsCleaner is an optional interface that a detector can implement to customize how its generated results
+// are "cleaned," which is defined as removing superfluous results from those found in a given chunk. The default
+// implementation of this logic removes all unverified results if there are any verified results, and all unverified
+// results except for one otherwise, but this interface allows a detector to specify different logic. (This logic must
+// be implemented outside results generation because there are circumstances under which the engine should not execute
+// it.)
+type CustomResultsCleaner interface {
+	// CleanResults removes "superfluous" results from a result set (where the definition of "superfluous" is detector-
+	// specific).
+	CleanResults(results []Result) []Result
+	// ShouldCleanResultsIrrespectiveOfConfiguration allows a custom cleaner to instruct the engine to ignore
+	// user-provided configuration that controls whether results are cleaned. (User-provided configuration is not the
+	// only factor that determines whether the engine runs cleaning logic.)
+	ShouldCleanResultsIrrespectiveOfConfiguration() bool
```

Comment:
Do you think this is something that would happen often?

Reply:
It's needed for `aws` and `awssessionkey` today.

Reply:
Hm.. what are your thoughts on special-casing `aws` and `awssessionkey` detectors instead of making this a required method for the interface?

Reply:
I am extremely opposed to special-casing any detectors in the code. My position is that if we've created a bad abstraction, we should own that, rather than splitting it into multiple parts with invisible coupling just so that we don't have to look at the mess we've made.